### ISSUE TITLE
Failed to match sequence (WHITESPACE? DIAGRAM_DEFINITION) at line 1 char 1. (Parslet::ParseFailed)

### ISSUE THREAD

**Issue by CAMOBAP** [lutaml/lutaml#19]
Samples and details of the crash can be obtained from:
 - https://github.com/metanorma/metanorma-model-un/actions/runs/657960651

cc @ronaldtse

**Comment by CAMOBAP** [comment]
@ronaldtse could you please help to assign this to the right person

**Comment by ronaldtse** [comment]
This is for @w00lf but likely each repo has individual problems.

**Comment by w00lf** [comment]
> Samples and details of the crash can be obtained from:
> 
> * https://github.com/metanorma/metanorma-model-un/actions/runs/657960651
> * https://github.com/metanorma/metanorma-model-mpfa/runs/2122809505
> * https://github.com/metanorma/metanorma-model-gb/runs/2122584596
> 
> cc @ronaldtse

hi there @CAMOBAP, lutaml-wsd2uml is not quite production-ready yet, it can create broken lutaml code and should be fixed manually after. That's your case. For example, models/OfferingDoc_DataTypes.wsd is transformed into this file:

```
diagram OfferingDoc_DataTypes {
  
  
  
  class Uri <<dataType>> {
  }
  
  class String <<primitiveType>> {
  }
  
  class Iso8601DateTime <<dataType>> {
  }
  
  class Iso15924Code <<dataType>> {
  }
  
  class Iso639Code <<dataType>> {
  }
  
  class LocalizedString <<StandardDocument>> {
  }
  
  
  
  Uri -[hidden]- String
  String -[hidden]- Iso8601DateTime
  Iso8601DateTime -[hidden]- Iso15924Code
  Iso15924Code -[hidden]- Iso639Code
  
}
```
The last 4 lines are completely wrong in lutaml. I can try to fix this issues now, but there is too much edge cases with wsd to lutaml transformation

**Comment by CAMOBAP** [comment]
@w00lf if you can give me a direction to fix I  think I can proceed on my own, I'm a bit familiar with `plantuml` syntax but `lutaml` is 'hark horse' for me)

**Comment by w00lf** [comment]
> @w00lf if you can give me a direction to fix I think I can proceed on my own, I'm a bit familiar with `plantuml` syntax but `lutaml` is 'hark horse' for me)

Well, in case of GbDoc-GbStandardPrefix.lutaml its because of `-[hidden]- ` notation in plantuml, `wsd2uml` does not support that kind of relations, same with lutaml, you can try to swap ot for the simple relation(eg `<|-`).

**Comment by CAMOBAP** [comment]
Now it happens on all models repos https://github.com/metanorma/metanorma-requirements-models/pull/19/checks?check_run_id=2940022599

@w00lf maybe it's related to some another issue

**Comment by w00lf** [comment]
> Now it happens on all models repos:
> 
> * https://github.com/metanorma/metanorma-requirements-models/pull/19/checks?check_run_id=2940022599
> * [metanorma/metanorma-build-scripts#66 Drop ruby 2.4 support metanorma/metanorma-model-gb#31](https://github.com/metanorma/metanorma-model-gb/pull/31)
> * [metanorma/metanorma-build-scripts#66 Drop ruby 2.4 support metanorma/metanorma-model-iso#67](https://github.com/metanorma/metanorma-model-iso/pull/67)
> * [metanorma/metanorma-build-scripts#66 Drop ruby 2.4 support metanorma/metanorma-model-standoc#33](https://github.com/metanorma/metanorma-model-standoc/pull/33)
> * [metanorma/metanorma-build-scripts#66 Drop ruby 2.4 support metanorma/metanorma-model-un#24](https://github.com/metanorma/metanorma-model-un/pull/24)
> * [metanorma/metanorma-build-scripts#66 Drop ruby 2.4 support metanorma/metanorma-model-mpfa#18](https://github.com/metanorma/metanorma-model-mpfa/pull/18)
> * [metanorma/metanorma-build-scripts#66 Drop ruby 2.4 support metanorma/metanorma-model-itu#20](https://github.com/metanorma/metanorma-model-itu/pull/20)
> * [metanorma/metanorma-build-scripts#66 Drop ruby 2.4 support metanorma/metanorma-model-csa#17](https://github.com/metanorma/metanorma-model-csa/pull/17)
> * [metanorma/metanorma-build-scripts#66 Drop ruby 2.4 support metanorma/metanorma-model-nist#34](https://github.com/metanorma/metanorma-model-nist/pull/34)
> * [metanorma/metanorma-build-scripts#66 Drop ruby 2.4 support metanorma/metanorma-model-iho#11](https://github.com/metanorma/metanorma-model-iho/pull/11)
> * [metanorma/metanorma-build-scripts#66 Drop ruby 2.4 support metanorma/metanorma-model-cc#26](https://github.com/metanorma/metanorma-model-cc/pull/26)
> * [metanorma/metanorma-build-scripts#66 Drop ruby 2.4 support metanorma/metanorma-model-m3d#20](https://github.com/metanorma/metanorma-model-m3d/pull/20)
> * [metanorma/metanorma-build-scripts#66 Drop ruby 2.4 support metanorma/metanorma-model-ogc#21](https://github.com/metanorma/metanorma-model-ogc/pull/21)
> * [metanorma/metanorma-build-scripts#66 Drop ruby 2.4 support metanorma/metanorma-model-rsd#19](https://github.com/metanorma/metanorma-model-rsd/pull/19)
> * [metanorma/metanorma-build-scripts#66 Drop ruby 2.4 support metanorma/metanorma-model-bipm#10](https://github.com/metanorma/metanorma-model-bipm/pull/10)
> * [metanorma/metanorma-build-scripts#66 Drop ruby 2.4 support metanorma/metanorma-model-ieee#9](https://github.com/metanorma/metanorma-model-ieee/pull/9)
> * [metanorma/metanorma-build-scripts#66 Drop ruby 2.4 support metanorma/basicdoc-models#28](https://github.com/metanorma/basicdoc-models/pull/28)
> * [metanorma/metanorma-build-scripts#66 Drop ruby 2.4 support metanorma/metanorma-requirements-models#19](https://github.com/metanorma/metanorma-requirements-models/pull/19)
> 
> @w00lf maybe it's related to some another issue

When did it start failing?

**Comment by CAMOBAP** [comment]
@w00lf to be honest IDK, I observed this during CI config update

@ronaldtse I propose to add cron to model repos, to get acknowledged about those issues earlier, how do you think?

**Comment by w00lf** [comment]
@CAMOBAP i have looked into metanorma-requirements-models repo, wsd files has comas at the end of each line - lutaml cannot handle that

**Comment by w00lf** [comment]
Actually there is a number of issues with the wsd files:

`Resolving dependencies...` message gets added at the start of the file
`together {` macro - we dont support it

**Comment by CAMOBAP** [comment]
@w00lf I apologize I still not very familiar with this part, what does this mean. should we fix wsd files to avoid this, or something else?

**Comment by w00lf** [comment]
> @w00lf I apologize I still not very familiar with this part, what does this mean. should we fix wsd files to avoid this, or something else?

Yes, you need to remove all `together` blocks from wsd files, `Resolving dependencies...` i can check myself, seems like some new behavier

**Comment by w00lf** [comment]
@CAMOBAP ok, `Resolving dependencies...` message was because i was not using `bundle exec`. In adition to `together` blocks wsd files also have comas at the end of each line, we should delete them too

**Comment by w00lf** [comment]
![image](https://user-images.githubusercontent.com/1158036/124269774-000c5780-db44-11eb-8841-43000fcccfda.png)
